src/bentoml_cli/auth_server.py

```diff
@@ -0,0 +1,156 @@
+    """
+    HTTPRequest Handler that is intended to be used as oauth2 callback page
+    """
+
+    def log_message(self, format, *args):
+        """
+        Disables logging for BaseHTTPRequestHandler
+        """
+        # silence the log messages
+        pass
+
+    # # noinspection PyPep8Naming
+    # def do_GET(self) -> None:
+    #     """
+    #     Provides callback page for the oauth2 redirect
+    #     """
+    #     try:
+    #         params = parse_qs(urlparse(self.path).query)
+
+    #         has_error = ("code" not in params
+    #                     or len(params['code']) != 1
+    #                     or params['code'][0].strip() == "")
+
+    #         if has_error:
+    #             self.send_response(400, "Something went wrong trying to authenticate you. Please try going back in your browser, or restart the auth process.")
+    #             self.end_headers()
+    #         else:
+    #             self.send_response(200, "You have been authenticated successfully. You may close this browser window now and go back to the terminal.")
+    #             self.end_headers()
+    #             self.server._code = params["code"][0]
+    #     except:
+    #         self.send_response(500)
+    #         self.end_headers()
+
+    def do_POST(self) -> None:
+        """
+        Provides callback page for the oauth2 redirect using POST request
+        """
+        try:
+            content_length = int(self.headers["Content-Length"])
+            post_data = self.rfile.read(content_length).decode("utf-8")
+            params = parse_qs(post_data)
+
+            has_error = (
+                "code" not in params
+                or len(params["code"]) != 1
+                or params["code"][0].strip() == ""
+            )
+
+            if has_error:
+                self.send_response(400)
+                self.end_headers()
+            else:
+                self.send_response(200)
+                self.end_headers()
+                self.server._code = params["code"][0]
+        except Exception:
+            self.send_response(500)
+            self.end_headers()
+
+
+class AuthCallbackHttpServer(HTTPServer):
+    """
+    Simplistic HTTP Server to provide local callback URL for oauth2 provider
+    """
+
+    def __init__(self, port):
+        super().__init__(("", port), AuthRedirectHandler)
+
+        self._code: str | None = None
+
+    def get_code(self) -> str | None:
+        """
+        This method should only be called after the request was done and might be None when no token is given.
+
+        :return: Authorization code or None if the request was not performed yet
+        """
+        return self._code
+
+    @property
+    def callback_url(self) -> str:
+        """
+        Callback URL for the HTTP-Server
+        """
+        return f"http://localhost:{self.server_port}"
+
+    def wait_for_code(self, attempts: int = 3, timeout_per_attempt=10) -> Optional[str]:
```

Comment:
Is it used anywhere?

If it is removed `_method_with_timeout` can also be removed.

Reply:
No. will delete it